src/nvim/ex_docmd.c

```diff
@@ -5890,6 +5921,20 @@ static void do_ucmd(exarg_T *eap)
   else
     cmd = USER_CMD_GA(&curbuf->b_ucmds, eap->useridx);
 
+  if (cmd->uc_argt & EX_LUA_CB) {
+    lua_State* L = executor_enter_lua();
+    nlua_pushref(L, cmd->lua_cb);
+    lua_pushlstring(L, (const char*)eap->arg, STRLEN(eap->arg));
+    lua_newtable(L);
+    if (eap->forceit == 1) {
+      lua_pushliteral(L, "bang");
+      lua_pushboolean(L, eap->forceit == 1);
+      lua_rawset(L, -3);
```

Comment:
can use `lua_setfield()` here (it invokes metamethods, but is safe for a freshly created table)

Reply:
Why not put this logic in `lua/executor.c` instead of exposing `executor_enter_lua` ?

Reply:
It depends on how many places I expected to need lua. With bfredl's recommendation for using `executor_free_luaref()`, I would only need to use `executor_enter_lua()` once now, which means I could replace it with a single function.

Architecturally, I don't find exposing the ability to do arbitrary Lua elsewhere to be terrible as long as it's used responsibly if it was the choice between that and a bunch of little utility functions.

In this case I think it's fine, but I'm also looking at adding Lua functions for keymappings and autocmds (I know there's an existing PR out there which I'm hoping to pick up). So I'll say for sure after spending more time with that.

But good point, nonetheless.

Reply:
To me, a "bunch of functions" in `lua/executor.c` seems like a better tradeoff than managing Lua state all over the place. That keeps `lua/executor.c` encapsulated.